File: JAVA/LowerCaseFilter.java
public class LowerCaseFilter extends DataPreprocessor {
    @Override
    public void preprocess(String data) {
        System.out.println("Preprocessing...Converting data to lowercase");
        // Can be done outside as well
        setNext(new RemoveStopWordsFilter());
        if (nextPreprocessor != null) {
            nextPreprocessor.preprocess(data);
        }
    }
}

File: C++/lowercasefilter.cpp
#include "lowercasefilter.hpp"

void LowerCaseFilter::preprocess(string data) {
    cout<<"Preprocessing...Converting data to lowercase"<<endl;
    //can be done outside as well
    setNext(new RemoveStopWordsFilter());
    if (nextPreprocessor) {
        nextPreprocessor->preprocess(data);
    }
}
